packages/aws-cdk-lib/aws-lambda-event-sources/lib/stream.ts

```diff
@@ -137,6 +137,33 @@ export interface StreamEventSourceProps extends BaseStreamEventSourceProps {
    */
   readonly filterEncryption?: IKey;
 
+  /**
+   * Configuration for enhanced monitoring metrics collection
+   * When specified, enables collection of additional metrics for the stream event source
+   *
+   * @default - Enhanced monitoring is disabled
+   */
+  readonly metricsConfig?: MetricsConfig;
+
+}
+
+export enum MetricType {
+  /**
+   * Event Count metrics provide insights into the processing behavior of your event source mapping,
+   * including the number of events successfully processed, filtered out, or dropped.
+   * These metrics help you monitor the flow and status of events through your event source mapping.
+   */
+  EVENT_COUNT = 'EventCount',
+}
+
+/**
+ * Configuration for collecting metrics from the event source
+ */
+export interface MetricsConfig {
+  /**
+  * List of metric types to enable for this event source
+  */
+  readonly metrics: MetricType[];
```

Comment:
Can we import these definition from event-source-mapping.ts instead of create a duplicate definition here?

Reply:
Sure, looking into it now